src/main/java/org/takes/facets/auth/PsChain.java

```diff
@@ -66,16 +64,18 @@ public PsChain(final Iterable<Pass> list) {
     }
 
     @Override
-    public Iterator<Identity> enter(final Request req) throws IOException {
-        final Collection<Identity> users = new ArrayList<Identity>(1);
+    public Opt<Identity> enter(final Request req) throws IOException {
+        Opt<Identity> retValue = null;
```

Comment:
Ok, I will rename that, but I don't agree with the ideia of initializing the variable. IMHO, this can be a waste of process and use of memory if this is going to be initialized again.